website/src/components/quickstartTOC/index.js

```diff
@@ -0,0 +1,223 @@
+// This component is used to build the functionality of the quickstart guides
+// Each H2 (##) in the markdown file is a step in the guide
+
+import React, { useState, useEffect } from "react";
+import clsx from "clsx";
+import style from "./styles.module.css";
+import { useLocation, useHistory } from "@docusaurus/router";
+import queryString from "query-string";
+
+function QuickstartTOC() {
+  const history = useHistory();
+  const location = useLocation()
+  const locationPath = useLocation().pathname;
+  const queryParams = queryString.parse(location.search);
+
+  const activeStepLocal = typeof localStorage !== "undefined" ? localStorage.getItem(locationPath) : 1;
+  const activeStepParam = queryParams.step ? (queryParams.step) : activeStepLocal ? activeStepLocal : 1;
+
+  const [mounted, setMounted] = useState(false);
+  const [tocData, setTocData] = useState([]);
+  const [activeStep, setActiveStep] = useState(activeStepLocal || 1);
+  const [activeQuickstart, setActiveQuickstart] = useState(locationPath);
+
+  useEffect(() => {
+    // Get all h2 for each step in the guide
+    const quickstartContainer = document.querySelector(".quickstart-container");
+    const steps = quickstartContainer.querySelectorAll("h2");
+    const snippetContainer = document.querySelectorAll(".snippet");
+
+    // Add snippet container to its parent step
+    snippetContainer.forEach((snippet) => {
+      const parent = snippet?.parentNode;
+      while (snippet?.firstChild && parent.className) {
+        if (parent) {
+          parent.insertBefore(snippet.firstChild, snippet);
+        }
+      }
+    });
+
+    // Create an array of objects with the id and title of each step
+    const data = Array.from(steps).map((step, index) => ({
+      id: step.id,
+      title: step.innerText,
+      stepNumber: index + 1,
+    }));
+
+    setTocData(data);
+    setMounted(true);
+    setActiveStep(activeStepParam || parseInt(activeStepLocal) || 1);
+
+    // Wrap all h2 (steps), along with all of their direct siblings, in a div until the next h2
```

Comment:
Nice work commenting on all functionality, definitely helps make everything easier to understand what's going on. Lots happening in this file, nice work structuring everything.